ext/hash/config.m4

```diff
@@ -24,25 +24,33 @@ if test "$PHP_HASH" != "no"; then
   AC_CHECK_SIZEOF(int, 4)
   AC_CHECK_SIZEOF(long, 4)
   AC_CHECK_SIZEOF(long long, 8)
-  
-  PHP_CHECK_64BIT([
-    SHA3_DIR="sha3/generic32lc"
-    SHA3_OPT_SRC="$SHA3_DIR/KeccakP-1600-inplace32BI.c"
-  ],[
-    SHA3_DIR="sha3/generic64lc"
-    SHA3_OPT_SRC="$SHA3_DIR/KeccakP-1600-opt64.c"
-  ])
-  EXT_HASH_SHA3_SOURCES="$SHA3_OPT_SRC $SHA3_DIR/KeccakHash.c $SHA3_DIR/KeccakSponge.c"
-  PHP_HASH_CFLAGS="-I@ext_srcdir@/$SHA3_DIR -DKeccakP200_excluded -DKeccakP400_excluded -DKeccakP800_excluded"
+
+  if test $ac_cv_c_bigendian_php = yes; then
```

Comment:
This over-pessimizes big endian.  32bit BE works fine on 32bitBE archs (well, I assume it does based on scanning the code).  BE is a minority regardless, so it's probably fine, but I'd be inclined to only go slow route for the broken path.

Reply:
I only have 64-bit bigendian target.... :(

Reply:
@sgolemon I think the safer way for now if to apply this PR (at least before RC4, so I can build 7.2 in Fedora), then we can work on it later... but I indeed think bigendian is minority, and 32-bit another minority... (we still have some for RHEL-6... 7 years old... and 7.2 already mostly unsupported there)

Reply:
I'm happy with that. 👍

Reply:
couldnt drop 32bit support be a great thing for php 7.3/8.0?

Reply:
@staabm I don't think we can drop 32-bit support, mostly because of ARM computers which are still quite common (aarch64 is not yet widely used)